### PR TITLE
Fix: Can't save 2 entities consecutively (React + manytomany)

### LINKED ISSUE #13182: Can't save 2 entities consecutively (React + manytomany)
##### **Overview of the issue**

Can't save 2 entities consecutively
Fails in the browser with 
```
Uncaught (in promise) TypeError: Cannot read property 'filter' of undefined
    at mapIdList (entity-utils.ts:23)
    at Object.saveEntity [as onSubmit] (foo-update.tsx:63)
    at AvForm._callee$ (AvForm.js:152)
    at tryCatch (runtime.js:62)
    at Generator.invoke [as _invoke] (runtime.js:296)
    at Generator.prototype.<computed> [as next] (runtime.js:114)
    at step (asyncToGenerator.js:17)
    at asyncToGenerator.js:28
```

##### **Reproduce the error**

Create a Foo, and immediately after, create a new one.
On 2nd save, it fails

##### **JHipster Version(s)**

```
clement-2@0.0.1-SNAPSHOT /Users/ctamisier/projects/clement2
└── (empty)

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "blueprints": [],
    "otherModules": [],
    "applicationType": "monolith",
    "baseName": "clement2",
    "jhipsterVersion": "6.10.5",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": [],
    "pages": [],
    "creationTimestamp": 1606926921414,
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "serviceDiscoveryType": false,
    "authenticationType": "jwt",
    "packageName": "com.mycompany.myapp",
    "serverPort": "8080",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "buildTool": "maven",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "react",
    "withAdminUi": true,
    "clientTheme": "none",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "packageFolder": "com/mycompany/myapp",
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "clientPackageManager": "npm",
    "clientThemeVariant": "",
    "languages": [
      "en"
    ],
    "reactive": false,
    "entities": [
      "Foo",
      "Bar"
    ],
    "lastLiquibaseTimestamp": 1606927729000
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Foo {
  code String
}
entity Bar {
  code String
}
relationship ManyToMany {
  Foo{bar} to Bar{foo}
}

paginate Foo, Bar with pagination
service Foo, Bar with serviceImpl

</pre>
</details>

##### **Environment and Tools**

openjdk version "11.0.7" 2020-04-14
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.7+10)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.7+10, mixed mode)

git version 2.29.2

node: v12.16.2

npm: 6.14.8

yeoman: 3.1.1

Docker version 19.03.13, build 4484c46d9d

docker-compose version 1.27.4, build 40524192

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #13182 

---

Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [x] Tests are added where necessary
-   [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: generators/client/templates/react/src/main/webapp/app/shared/util/entity-utils.ts.ejs [hunk 1]
```diff
@@ -38,7 +38,12 @@ export const cleanEntity = entity => {
  * @param idList Elements to map.
  * @returns The list of objects with mapped ids.
  */
- export const mapIdList = (idList: ReadonlyArray<any>) => idList.filter((entityId: any) => entityId !== '').map((entityId: any) => ({ id: entityId }));
+export const mapIdList = (idList: ReadonlyArray<any>) => {
```
Comment (pascalgrimaud) [inline generators/client/templates/react/src/main/webapp/app/shared/util/entity-utils.ts.ejs:41]: indent

**Comment by pascalgrimaud** [comment]
can you rewrite your commit text with more explicite ones plz ?

**Comment by ctamisier** [comment]
It's a detail but I think it's better to have only one commit.
For me the test is bound to the fix. If you checkout this commit (the fix), I would like to have the test as well.

File: generators/client/templates/react/src/main/webapp/app/shared/util/entity-utils.ts.ejs (hunk 1 above)
Comment (qmonmert) [reply generators/client/templates/react/src/main/webapp/app/shared/util/entity-utils.ts.ejs:41]: all the files have this indentation in `react/src/app`

**Comment by qmonmert** [comment]
@pascalgrimaud done (rewrite commit)
@ctamisier done (just one commit)

File: generators/client/templates/react/src/main/webapp/app/shared/util/entity-utils.ts.ejs (hunk 1 above)
Comment (pascalgrimaud) [reply generators/client/templates/react/src/main/webapp/app/shared/util/entity-utils.ts.ejs:41]: oh ok, the bad indent is the following function :D nsot the part you modified

**Review by ruddell — APPROVED** [review]
Tested and it fixes the issue, thanks @qmonmert

**Pull request merged**